src/app/loadout/fashion/FashionDrawer.tsx

```diff
@@ -508,34 +505,51 @@ function FashionSocket({
     unlockedPlugSetItemsSelector(state, storeId)
   );
   const handleOrnamentClick = socket && (() => onPickPlug({ item: exampleItem, socket }));
+
+  const unlockedPlugsWithoutTheDefault = Array.from(unlockedPlugSetItems).filter(
+    (plugHash) => plugHash !== defaultPlug.hash
+  );
+
   const canSlotOrnament =
-    plugHash !== undefined &&
-    (plugHash === socket?.emptyPlugItemHash ||
-      (unlockedPlugSetItems.has(plugHash) &&
-        socket?.plugSet?.plugs.some((p) => p.plugDef.hash === plugHash)) ||
-      socket?.reusablePlugItems?.some((p) => p.plugItemHash === plugHash && p.enabled));
+    (socket?.plugSet &&
+      _.intersection(
+        unlockedPlugsWithoutTheDefault,
+        socket.plugSet.plugs.map((plug) => plug.plugDef.hash)
+      ).length > 0) ||
+    (socket?.reusablePlugItems &&
+      _.intersection(
+        unlockedPlugsWithoutTheDefault,
+        socket.reusablePlugItems.filter((p) => p.enabled).map((p) => p.plugItemHash)
+      ).length > 0);
```

Comment:
Bit of a change in functionality but I think this is more correct for the purpose.

Basically, we are checking to see if there exists a plug, that is all of unlocked, not the default, and in one of the plug sets for the socket. If there is then we can assume an ornament/shader exists which can be socketed into the socket. Otherwise the socket doesn't exists (so no ornament) or no ornament that isn't the default exists, thus the user cannot socket anything.